js/main.js

```diff
@@ -22,23 +32,131 @@ var stockClose = [];
 var stockDateTime = [];
 $seachBtn.addEventListener('click', seachStock);
 $displayTable.addEventListener('click', displayTableClick);
+$displayWatchListTable.addEventListener('click', displayWatchListTableClick);
 $close.addEventListener('click', closeSpecificStock);
+$watchlistNavBtn.addEventListener('click', watchlistNavBtnClick);
+$homeBtn.addEventListener('click', homeBtnClick);
+$watchListBtn.addEventListener('click', watchListBtnClick);
+$deleteBtn.addEventListener('click', deleteBtnClick);
+$deleteBtnConfirm.addEventListener('click', deleteBtnConfirmClick);
+$cancelBtnModal.addEventListener('click', cancelBtnModalClick);
+
+function deleteBtnClick() {
+  $cancelationPage.className = 'cancelationPage';
+}
+function cancelBtnModalClick() {
+  $cancelationPage.className = 'cancelationPage hidden';
+}
+function deleteBtnConfirmClick() {
+  if (data.isWatchlist === true) {
+    $cancelationPage.className = 'cancelationPage hidden';
+    data.watchlistEntries.splice(data.watchlistEntries.indexOf(data.currentStock), 1);
+    data.currentStock = null;
+    data.isPortfolio = false;
+    data.isWatchlist = true;
+    switchView('watchlist-view');
+  }
+
+}
+
+function displayWatchListTableClick(event) {
+  if (event.target.getAttribute('watchlist-symbol') === null) {
+    return;
+  }
+  data.currentStock = event.target.getAttribute('watchlist-symbol');
+  switchView('stock-view');
+}
+
+function watchListBtnClick(event) {
+  if (data.watchlistEntries.indexOf(data.currentStock) === -1) {
+    data.watchlistEntries.push(data.currentStock);
+  }
+  data.currentStock = null;
+  data.isPortfolio = false;
+  data.isWatchlist = true;
```

Comment:
What is the purpose of these properties? Is it to know what the current view is? If so, you should just save the name of the current view into one property.